src/cmd/create.go

```diff
@@ -0,0 +1,490 @@
+	if err != nil {
+		logrus.Error(err)
+	}
+
+	dbusSystemBusAddressBind := []string{"--volume", fmt.Sprintf("%s:%s", dbusSystemBusAddress, dbusSystemBusAddress)}
+	createArgs = append(createArgs, dbusSystemBusAddressBind...)
+
+	logrus.Info("Preparing sssd-kcm socket")
+	args = []string{"show", "--value", "--property", "Listen", "sssd-kcm.socket"}
+	output, err := utils.SystemctlOutput(args...)
+	if err != nil {
+		logrus.Error("Failed to use 'systemctl show'")
+	}
+
+	kcmSocket = strings.Trim(string(output), "\n")
+
+	if kcmSocket == "" {
+		logrus.Error("Failed to read property Listen from sssd-kcm.socket")
+	} else {
+		logrus.Infof("Checking value %s of property Listen in sssd-kcm.socket", kcmSocket)
+		if !strings.Contains(kcmSocket, " (Stream)") {
+			kcmSocket = ""
+			logrus.Error("Unknown socket in sssd-kcm.socket\nExpected SOCK_STREAM")
+		}
+		if !strings.Contains(kcmSocket, "/") {
+			kcmSocket = ""
+			logrus.Error("Unknown socket in sssd-kcm.socket\nExpected file system socket in the AF_UNIX family")
+		}
+	}
+
+	logrus.Infof("Parsing value %s of property Listen in sssd-kcm.socket", kcmSocket)
+	if kcmSocket != "" {
+		kcmSocket = strings.TrimSuffix(kcmSocket, " (Stream)")
+		kcmSocketBind = []string{"--volume", fmt.Sprintf("%s:%s", kcmSocket, kcmSocket)}
+		createArgs = append(createArgs, kcmSocketBind...)
+	}
+
+	logrus.Info("Checking if 'podman create' supports option '--ulimit host'")
+	if podman.CheckVersion("1.5.0") {
+		logrus.Info("Option '--ulimit host' is supported")
+		ulimitHost = []string{"--ulimit", "host"}
+	} else {
+		logrus.Info("Option '--ulimit host' is not supported")
+	}
+
+	homeEnv := strings.Split(viper.GetString("HOME"), "=")[0]
+	homeCanonical, err = filepath.EvalSymlinks(homeEnv)
+	if err != nil {
+		logrus.Fatalf("Failed to canonicalize %s", homeEnv)
+	}
+	logrus.Infof("Canonicalized %s to %s", homeEnv, homeCanonical)
+
+	logrus.Info("Checking if /home is a symbolic link to /var/home")
+	homeSymPath, err := filepath.EvalSymlinks("/home")
+	if err != nil {
+		logrus.Error("Failed to evaluate if /home is a symbolic link")
+	}
+	if homeSymPath == "/var/home" {
+		logrus.Info("/home is a symbolic link to /var/home")
+		command = append(command, "--home-link")
+	}
+
+	conn, err := dbus.SessionBus()
+	if err != nil {
+		logrus.Error("Failed to connect to Session Bus")
+	}
+	defer conn.Close()
+
+	logrus.Info("Calling org.freedesktop.Flatpak.SessionHelper.RequestSession")
+	SessionHelper := conn.Object("org.freedesktop.Flatpak", "/org/freedesktop/Flatpak/SessionHelper")
+	call := SessionHelper.Call("org.freedesktop.Flatpak.SessionHelper.RequestSession", 0)
+	if call.Err != nil {
+		logrus.Fatal("Failed to call org.freedesktop.Flatpak.SessionHelper.RequestSession")
+	}
+
+	// Add the environment variables that hold a value
+	for _, env := range preservedEnvVars {
+		value := viper.GetString(env)
+		if len(value) != 0 {
+			createArgs = append(createArgs, fmt.Sprintf("--env=%s=%s", env, value))
```

Comment:
This is meant to be done only in `run`, not `create`. Or am I missing something?

Reply:
I think the duplication is also in Shell Toolbox. I kept it in. We can drop it.